src/compile/mark/mixins.ts

```diff
@@ -551,7 +551,7 @@ function position2orSize(channel: 'x2' | 'y2', markDef: MarkConfig) {
   if (markDef[channel]) {
     return {[channel]: ref.vgValueRef(channel, markDef[channel])};
   } else if (markDef[sizeChannel]) {
-    return {[sizeChannel]: markDef[sizeChannel]};
+    return {[sizeChannel]: {value: markDef[sizeChannel]}};
```

Comment:
I'm glad I asked you to write a test :-P

Reply:
lol the editor broken the other day and I was rushing to sleep.  This is also the cause of bug in the depending image PR.